### ISSUE TITLE
convenience functions to_geojson(), to_dataframe() and to_geodataframe()

### ISSUE THREAD

**Issue by davisrbr** [sentinelsat/sentinelsat#216]
I am receiving error messages when I try to convert to a pandas/geopandas dataframe. This code works and downloads my requested sentinel satellite images:

    from sentinelsat import SentinelAPI, read_geojson, geojson_to_wkt
    from datetime import date
    
    api = SentinelAPI('*****', '*****', 'https://scihub.copernicus.eu/dhus')
    
    footprint = geojson_to_wkt(read_geojson('testAPIpoly.geojson'))

    products = api.query(footprint, cloudcoverpercentage = (0,10))

    #this works  
    api.download_all(products)

However if I instead attempt to convert to a pandas dataframe

    #api.download_all(products)
    
    #this does not work
    products_df = api.to_dataframe(products)
    
    api.download_all(products_df)

I receive an extensive error message that includes 

> "sentinelsat.sentinel.SentinelAPIError: HTTP status 500 Internal Server Error: InvalidKeyException : Invalid key (title) to access Products
" 

(where (title) is also replaced with **processed**, **platformname**, **processingbaseline**, etc.). I've tried a few different ways to convert to a dataframe and filter/sort results and have received an error message every time.

**Comment by Fernerkundung** [comment]
The `download_all` method does not accept a DataFrame as input. However, you can use the IDs in the dataframe as a direct input to the download method.

This is covered in the documentation with an example on [sorting&filtering](https://sentinelsat.readthedocs.io/en/stable/api.html#sorting-filtering).

```python
# convert to Pandas DataFrame
products_df = api.to_dataframe(products)

# sort and limit to first 5 sorted products
products_df_sorted = products_df.sort_values(['cloudcoverpercentage', 'ingestiondate'], ascending=[True, True])
products_df_sorted = products_df_sorted.head(5)

# download sorted and reduced products
api.download_all(products_df_sorted['id'])
```

**Comment by davisrbr** [comment]
Thanks. I receive a different error message when I run the example on sorting&filtering out-of-the-box, concluding with: 

> KeyError: 'id'

**Comment by Fernerkundung** [comment]
You are correct, we failed to update the documentation and the example is wrong right now. Thanks for the report!

The change that caused this was that we returned the result of `query` as OrderedDict and used the DHuS UUID as DataFrame index, instead of populating the column `id`.

The correct workflow now is:

```python
# convert to Pandas DataFrame
products_df = api.to_dataframe(products)

# sort and limit to first 5 sorted products
products_df_sorted = products_df.sort_values(['cloudcoverpercentage', 'ingestiondate'], ascending=[True, True])
products_df_sorted = products_df_sorted.head(5)

# download sorted and reduced products
api.download_all(products_df_sorted.index)
```

**Comment by Fernerkundung** [comment]
Fixed with 279745ec7a9050927fac7ee44bf004619df90a1b

The example is now fixed in the master branch of the documentation: https://sentinelsat.readthedocs.io/en/master/api.html#sorting-filtering
